presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -239,8 +239,15 @@ private void refreshResourceGroupRuntimeInfo()
     {
         try {
             List<ResourceGroupRuntimeInfo> resourceGroupInfos = resourceGroupService.getResourceGroupInfo();
+            ImmutableMap<ResourceGroupId, ResourceGroupRuntimeInfo> map = resourceGroupInfos.stream().collect(toImmutableMap(ResourceGroupRuntimeInfo::getResourceGroupId, i -> i));
             resourceGroupRuntimeInfos.set(resourceGroupInfos.stream().collect(toImmutableMap(ResourceGroupRuntimeInfo::getResourceGroupId, i -> i)));
             lastUpdatedResourceGroupRuntimeInfo.set(currentTimeMillis());
+            boolean updatedSnapshot = updateResourceGroupsSnapshot();
+            for (RootInternalResourceGroup group : rootGroups) {
```

Comment:
Can we use lambda expression here and move the condition "if (updatedSnapshot)" outside of the iteration?